### PR TITLE
Count of error responses is not consistent based on SDK speed.

### PR DESCRIPTION
**Pull request by JamieSinn**

### CODE REVIEW

File: harness/features/initialize.local.test.ts [hunk 1]
```diff
@@ -105,7 +103,9 @@ describe('Initialize Tests - Local', () => {
             0,
         )
         expect((await variable.json()).data.value).toEqual(0)
-        await wait(1100)
+        scope.get(configRequestUrl).reply(200, testClient.getValidConfig)
```
Comment (JamieSinn) [inline harness/features/initialize.local.test.ts:106]: this apparently clears all previous and sets the new return scope. TIL

File: harness/features/initialize.local.test.ts [hunk 2]
```diff
@@ -88,15 +88,13 @@ describe('Initialize Tests - Local', () => {
         const testClient = new LocalTestClient(sdkName)
         const configRequestUrl = testClient.getValidConfigPath()
 
-        scope.get(configRequestUrl).times(2).reply(500)
+        // a 500 is a recoverable error so it will retry the connection and potentially make more requests than expected.
```
Comment (suthar26) [inline harness/features/initialize.local.test.ts:91]: why would it make more requests than the polling interval? or is the interval timer only started at successful call?

Comment (JamieSinn) [reply harness/features/initialize.local.test.ts:91]: Correct - the interval timer starts at the successful call.

Comment (suthar26) [reply harness/features/initialize.local.test.ts:91]: for all sdks? if that was the case would we have caught this or saw inconsistencies earlier?

Comment (suthar26) [reply harness/features/initialize.local.test.ts:91]: not fully confident that this wouldnt break other sdks.. but i guess if the tests pass then we can merge it..

**Comment by JamieSinn** [comment]
event flake - should be fine

**Pull request merged**